rundk/main.go

```diff
@@ -0,0 +1,132 @@
+	}
+
+	builtUpDefers := make([]func(), 0)
+
+	// Setup command
+	cmd := interactive.NewDocker()
+
+	// Setup args to promote env vars from local to the Docker container
+	envs := interactive.Env{}
+	if err = envs.PromoteFromEnv(mandatoryEnvVars...); err != nil {
+		log.Fatal("Missing mandatory argument: ", err)
+	}
+	envs.PromoteFromEnv(optionalEnvVars...) // Optional, so don't check error
+
+	gcloudKey := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+	// to the container
+	if gcloudKey != "" {
+		cmd.AddMount("bind", gcloudKey, gcloudKey)
+	}
+	// Mount the required files and directories on the host machine to the container
+	for _, m := range mounts {
+		if strings.TrimSpace(m) != "" {
+			cmd.AddMount("bind", m, m)
+		}
+	}
+
+	repoRoot, err := helpers.GetRootDir()
+	if err != nil {
+		log.Fatal("Error getting the repo's root directory: ", err)
+	}
+
+	// Setup temporary directory
+	tmpDir, err := ioutil.TempDir("/tmp", "prow-docker.")
+	if err != nil {
+		log.Fatal("Error setting up the temporary directory: ", err)
+	}
+
+	// Copy and mount source code dir
+	// Add overlay mount over the user's git repo, so the flow doesn't mess it
+	// up
+	cleanup, err := cmd.CopyAndAddMount("bind", tmpDir, repoRoot, filepath.Dir(repoRoot))
+	if err != nil {
+		log.Fatal("Error setting up mount of repo's root directory to the container: ", err)
+	}
+	builtUpDefers = append(builtUpDefers, cleanup)
+
+	// Copy and mount for kube context to be available (if reusing an existing cluster)
+	// If future use needs other directories, mounting the whole home directory could be a pain
+	//  because our prow-tests image will be installing Go in /root/.gvm
+	cleanup, err = cmd.CopyAndAddMount("bind", tmpDir, path.Join(os.Getenv("HOME"), ".kube"), "/root/.kube")
+	if err != nil {
+		log.Fatal("Error setting up mount of kubeconfig to the container: ", err)
+	}
+	builtUpDefers = append(builtUpDefers, cleanup)
+
+	cmd.LogFile = path.Join(tmpDir, "build-log.txt")
+	log.Print("Logging to ", cmd.LogFile)
+
+	extArtifacts := os.Getenv("ARTIFACTS")
+	// Artifacts directory
+	if extArtifacts == "" {
+		log.Print("Setting local ARTIFACTS directory to ", tmpDir)
+		extArtifacts = tmpDir
+	}
+	cmd.AddMount("bind", extArtifacts, extArtifacts)
+	envs["ARTIFACTS"] = extArtifacts
+	builtUpDefers = append(builtUpDefers, func() {
+		log.Print("💡 Artifacts can be found at ", extArtifacts)
+	})
+	cmd.AddEnv(envs)
+
+	// Starting directory
+	cmd.AddArgs("-w=" + repoRoot)
+
+	return cmd, func() {
+		for _, ff := range builtUpDefers {
+			if ff != nil {
+				ff()
+			}
+		}
+	}
+}
+
+func run(cmd interactive.Docker, image string, commandAndArgsOpt ...string) error {
+	// Finally add the image then command to run (if any)
+	cmd.AddArgs(image)
+	cmd.AddArgs("runner.sh")
+	cmd.AddArgs(commandAndArgsOpt...)
+	log.Println(cmd)
+	log.Println("👆 Starting in 3 seconds, ^C to abort!")
```

Comment:
If this is an important step, I would suggest to expand this to 5+ seconds. Reading the instruction will take ~1 sec already and we need to give room for the reaction from the user

Reply:
It's not a very important step, and also since users could be running this tool again and again, I feel 5+ seconds is a bit too long, so I'm inclined to keep it as 3 seconds.

Reply:
The purpose is: what if you just want the env setup and to use it yourself; then 3 seconds is fast enough because you'll be watching it.
Then why not wait on user input? I hate when you want to run something which takes a while, you launch it, walk away, then it doesn't start because of unnecessary user input.

Reply:
I do agree 3 seconds might be enough for a user who is used to invoking the tool, but it might leave ~1-2 seconds of reaction to a developer who does read the message for the first time.
In any case, being the outcome controlled I'm fine with either option.

+1 to @coryrc 's comment. Waiting on user's input does not seem to be the best option at all.